re.c

```diff
@@ -3829,6 +3825,10 @@ rb_reg_initialize_m(int argc, VALUE *argv, VALUE self)
         rb_get_kwargs(kwargs, keywords, 0, 1, &timeout);
     }
 
+    if (argc == (rb_keyword_given_p() ? 4 : 3)) {
```

Comment:
`rb_scan_args` returns the number of arguments excluding keyword argument hash.

Reply:
@nobu Thanks, used that approach in latest commit.  Also removed the optparse change in latest commit, since you fixed optparse.